File: backend/src/main/java/com/urbanman/ecommerce/model/Review.java
package com.urbanman.ecommerce.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "reviews")
public class Review {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long reviewId;

    @ManyToOne @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private String userName;     // snapshot at review time

    @Column(nullable = false)
    private Integer rating;      // 1 to 5

    @Column(length = 1000)
    private String comment;

    private LocalDateTime createdAt = LocalDateTime.now();

    // Getters & Setters
    public Long getReviewId()                       { return reviewId; }
    public void setReviewId(Long id)                { this.reviewId = id; }

    public Product getProduct()                     { return product; }
    public void setProduct(Product product)         { this.product = product; }

    public User getUser()                           { return user; }
    public void setUser(User user)                  { this.user = user; }

    public String getUserName()                     { return userName; }
    public void setUserName(String userName)        { this.userName = userName; }

    public Integer getRating()                      { return rating; }
    public void setRating(Integer rating)           { this.rating = rating; }

    public String getComment()                      { return comment; }
    public void setComment(String comment)          { this.comment = comment; }

    public LocalDateTime getCreatedAt()             { return createdAt; }
    public void setCreatedAt(LocalDateTime t)       { this.createdAt = t; }
}

File: backend/models/Review.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');
const User = require('./User');
const Product = require('./Product');

const Review = sequelize.define('Review', {
  reviewId: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true,
    field: 'review_id'
  },
  productId: {
    type: DataTypes.BIGINT,
    allowNull: false,
    field: 'product_id',
    references: {
      model: Product,
      key: 'product_id'
    }
  },
  userId: {
    type: DataTypes.BIGINT,
    allowNull: false,
    field: 'user_id',
    references: {
      model: User,
      key: 'user_id'
    }
  },
  userName: {
    type: DataTypes.STRING,
    allowNull: true,
    field: 'user_name'
  },
  rating: {
    type: DataTypes.INTEGER,
    allowNull: false,
    validate: {
      min: 1,
      max: 5
    }
  },
  comment: {
    type: DataTypes.STRING(1000),
    allowNull: true
  },
  createdAt: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW,
    field: 'created_at'
  }
}, {
  tableName: 'reviews',
  timestamps: false
});

Review.belongsTo(User, { foreignKey: 'userId', as: 'user' });
Review.belongsTo(Product, { foreignKey: 'productId', as: 'product' });

module.exports = Review;
